mirror-service/src/main/java/org/hyades/osv/OsvToCyclonedxParser.java

```diff
@@ -36,156 +36,164 @@
 public class OsvToCyclonedxParser {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(OsvToCyclonedxParser.class);
-    final Bom cyclonedxBom;
-    String ecosystem;
-    Vulnerability.Rating.Severity severity = Vulnerability.Rating.Severity.UNKNOWN;
+    private static final String SEVERITY = "severity";
+    private static final String DATABASE_SPECIFIC = "database_specific";
 
-    public OsvToCyclonedxParser() {
-        this.cyclonedxBom = new Bom();
-        this.ecosystem = null;
-    }
-
-    public Bom parse(final JSONObject object) {
+    public static Bom parse(JSONObject object) {
+        var cyclonedxBom = new Bom();
+        var severity = Vulnerability.Rating.Severity.UNKNOWN;
 
         // initial check if advisory is valid or withdrawn
         String withdrawn = object.optString("withdrawn", null);
+        if (withdrawn != null) {
```

Comment:
do we not need null check on the object itself too? before trying to access the object at line 47 to avoid NPE.

Reply:
Before this method is called, a new object is created 
            var json = new JSONObject(out.toString());
            Bom osvAdvisory = OsvToCyclonedxParser.parse(json);

We will not get null here I think